src/containers/Home.js

```diff
@@ -1,59 +1,151 @@
-import React, { useEffect } from "react";
+import React, { useCallback, useEffect } from "react";
 import { useState } from "react";
 import { useSelector } from "react-redux";
 import { useDispatch } from "react-redux";
-import { getCity, getWeather } from "../store/actions/weather";
+import {
+  clearWeatherData,
+  getCity,
+  getWeather,
+  setSelectedCity,
+} from "../store/actions/weather";
 import City from "../components/City.js";
 import DayCard from "../components/DayCard.js";
 import Search from "../components/Search.js";
 import "../styles/Home.css";
 import { getDaysWithHourlyDetails } from "../constants";
 import { getDateFromMs } from "../constants/dateFunctions";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+import CityCard from "../components/CityCard";
 
 function Home() {
   const dispatch = useDispatch();
+
   const weather = useSelector((state) => state.weather);
   const city = useSelector((state) => state.city);
+  const selectedCity = useSelector((state) => state.selectedCity);
+  const searchedTerm = useSelector((state) => state.searchedTerm);
 
   const [weatherData, setWeatherData] = useState(weather);
-  const [currentCity, setCurrentCity] = useState(city);
+  const [currentCity, setCurrentCity] = useState(selectedCity);
+  const [searchTerm, setSearchTerm] = useState(searchedTerm);
+
+  const [cityElements, setCityElements] = useState([]);
+  const [dayArrayElements, setDayArrayElements] = useState([]);
+
+  // Modal open state
+  const [modal, setModal] = React.useState(false);
```

Comment:
Why React.useState? you have already import the useState then use the imported one. Also the state name should reflect that it is used for modal open state. Example `[isModalOpen, setIsModalOpen]`

Reply:
changed